### PR TITLE
Integration2 t6f2 91 reviews refurbish

### PR DESCRIPTION
**Pull request by bovornc**
Refactored reviews to be standalone (instead of being a tuit)

### CODE REVIEW

**Review by nala-he — APPROVED** [review]
Tested locally on the postman. I think it is good! Thank you!

File: controllers/ReviewController.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        }
+        return ReviewController.reviewController;
+    }
+    private constructor() {}
+
+    /**
+     * Creates a new review instance
+     * @param {Request} req Represents request from client, including path
+     * parameter criticid identifying the primary key of the user that created
+     * the review, reviewid identifyng the primary key of the restaurant being reviewed
```
Comment (MaceeQi) [inline controllers/ReviewController.ts]: Parameter name is actually restaurantid and not reviewid - do you mind updating this? And there's also a small typo in "identifying" right after reviewid

File: controllers/ReviewController.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    findAllReviews = (req: Request, res: Response) =>
+        ReviewController.reviewDao
+            .findAllReviews()
+            .then(reviews => res.json(reviews));
+
+    /**
+     * Retrieves all reviews from the database by a particular critic and returns
+     * an array of reviews
+     * @param {Request} req Represents request from client, including path
+     * parameter reviewid identifying the primary key of the critic to retrieve
```
Comment (MaceeQi) [inline controllers/ReviewController.ts]: reviewid here should be criticid

File: controllers/ReviewController.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    findAllReviewsByCritic = (req: Request, res: Response) =>
+        ReviewController.reviewDao
+            .findAllReviewsByCritic(req.params.criticid)
+            .then(reviews => res.json(reviews));
+
+    /**
+     * Retrieves all reviews from the database for a particular restaurant and returns
+     * an array of reviews
+     * @param {Request} req Represents request from client, including path
+     * parameter restaurant identifying the primary key of the restaurant to retrieve
```
Comment (MaceeQi) [inline controllers/ReviewController.ts]: path parameter is restaurantid

File: controllers/ReviewController.ts [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    deleteReview = (req: Request, res: Response) =>
+        ReviewController.reviewDao
+            .deleteReview(req.params.reviewid)
+            .then(review => res.json(review));
+
+    /**
+     * Retrieves all reviews from the database and returns an array of reviews
+     * @param {Request} req Represents request from client
+     * @param {Response} res Represents response to client, including the
+     * body formatted as JSON arrays containing the eviews objects
```
Comment (MaceeQi) [inline controllers/ReviewController.ts]: typo - reviews

**Review by MaceeQi — COMMENTED** [review]
Tested on Postman and everything works! Just some small updates needed in JSDoc but otherwise, looks good to merge :)

**Comment by bovornc** [comment]
Fixed JSDoc typos

**Review by t1yena — APPROVED** [review]
tested locally and everything looks good! The jsdoc changes requested by macee looks good too. I think we can merge it now!

**Pull request merged**